### DISCUSSION TITLE
problem with importer.py - unable to read database

### DISCUSSION THREAD

**Discussion by juxtapositions-g** [qutebrowser/qutebrowser#8137] [Q&A] [answered]
so i'm trying to migrate my bookmarks from firefox to qutebrowser, exported them as html and:
❯ ./importer.py -i mozilla ~/firef_bookmarks_22-03-2024.html 

gives this output:
Traceback (most recent call last):
  File "/home/juxt/qutebrowser-bookmarks-importer/./importer.py", line 366, in <module>
    main()
  File "/home/juxt/qutebrowser-bookmarks-importer/./importer.py", line 93, in main
    import_function[input_format](args.bookmarks, bookmark_types,
  File "/home/juxt/qutebrowser-bookmarks-importer/./importer.py", line 308, in import_moz_places
    places = sqlite3.connect(os.path.join(profile, "places.sqlite"))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: unable to open database file

how to fix? what's wrong?

**Comment by The-Compiler** [comment, marked as answer]
From `--help`:

> bookmarks             Bookmarks file (html format) or profile folder (Mozilla format)

Right now, you're pointing the script to a HTML file, but are telling it that it's a firefox profile folder.

**Reply by juxtapositions-g** [reply]
i see. so how should i word my command the right way? thank you for your help!

**Reply by The-Compiler** [reply]
Either point it to the Firefox profile directory, or use `html` as import format instead of `mozilla`.

**Reply by juxtapositions-g** [reply]
i'll try it later today. thank you very much!

**Reply by juxtapositions-g** [reply]
hey! so now it says a similar thing, except that "no module named bs4"

**Reply by juxtapositions-g** [reply]
❯ ./importer.py -b ~/firef_bookmarks_22-03-2024.html   
Traceback (most recent call last):
  File "/home/juxt/qutebrowser-bookmarks-importer/./importer.py", line 366, in <module>
    main()
  File "/home/juxt/qutebrowser-bookmarks-importer/./importer.py", line 93, in main
    import_function[input_format](args.bookmarks, bookmark_types,
  File "/home/juxt/qutebrowser-bookmarks-importer/./importer.py", line 208, in import_netscape_bookmarks
    import bs4
ModuleNotFoundError: No module named 'bs4'

**Reply by juxtapositions-g** [reply]
sorry, can't figure that out
and when I try to point it to .mozilla/firefox folder, it goes:

> ./importer.py firefox ~/.mozilla/firefox               
Traceback (most recent call last):
  File "/home/juxt/qutebrowser-bookmarks-importer/./importer.py", line 366, in <module>
    main()
  File "/home/juxt/qutebrowser-bookmarks-importer/./importer.py", line 93, in main
    import_function[input_format](args.bookmarks, bookmark_types,
  File "/home/juxt/qutebrowser-bookmarks-importer/./importer.py", line 313, in import_moz_places
    c.execute(place_query[typ])
sqlite3.OperationalError: no such table: moz_bookmarks

**Reply by juxtapositions-g** [reply]
okay sorry to pester you, but I almost figured it out (i thought) after providing it with the right folder, but then it just gave a huge list of my bookmarks as output in my terminal. i checked, bookmarks in qutebrowser are nowhere to be seen. 
i gave it this command: ❯ ./importer.py firefox ~/.mozilla/firefox/6x36s3kk.default-esr 
are there some args i should also add or...?

**Reply by The-Compiler** [reply]
As for the first error, you will need to install [beautifulsoup4](https://pypi.org/project/beautifulsoup4/) as a dependency.

As for it printing the output, once again, see its `--help` which explains how to use it:

> Redirect the output from this script to the appropriate file in your
> qutebrowser config directory (listed in the output of :version),
> usually done with the '>' operator; for example,
> 
>     ./importer.py -i mozilla your_profile_path > ~/.config/qutebrowser/quickmarks

**Reply by juxtapositions-g** [reply]
heyyyyyyy it worked! thanks so much for your help!!!!!

**Reply by juxtapositions-g** [reply]
hmm. i don't think it imported all of them (i have a lot) and i wonder why...

**Reply by The-Compiler** [reply]
Hmm, no idea off-hand. Did you end up using the Firefox format? Maybe HTML works better?

**Reply by juxtapositions-g** [reply]
yeah, i used the mozilla format.

./importer.py -i mozilla ~/.mozilla/firefox/6x36s3kk.default-esr > ~/.config/qutebrowser/quickmarks

**Reply by juxtapositions-g** [reply]
it's not crucial i'll try some other time. maybe this has something to do with the fact that i have a lot of folders (including nested inside one another) and stuff... since quickmarks don't have that, i'm not sure, but maybe it imported the bookmarks that were just hanging there unsorted. it's just a suggestion hehe i'll check my theory sometime (it's tedious since i have a huge amount of bookmarks)

**Reply by juxtapositions-g** [reply]
thank you so much for your help and for being so patient :) appreciate that!
